src/robusta/integrations/mattermost/client.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from http import HTTPStatus
+from typing import Optional, List, Dict
+
+from ..common.requests import process_request, HttpMethod
+from ...core.model.env_vars import ROBUSTA_LOGO_URL
+
+_SUPPORTED_SCHEMAS = ['http', 'https']
+_API_PREFIX = "api/v4"
+
+
+class MattermostClient:
+    def __init__(
+            self, url: str, token: str, token_id: str, channel_id: str, schema: Optional[str]
+    ):
+        """
+        Set the Mattermost webhook url.
+        """
+        self.client_url = url.split(":")[-1].lstrip("/").rstrip("/")
+        self.token = token
+        self.token_id = token_id
+        self.channel_id = channel_id
+        schema = schema or "https"
+        if schema not in _SUPPORTED_SCHEMAS:
+            raise AttributeError(f"{schema} is not supported schema, should be one of following: {_SUPPORTED_SCHEMAS}")
+        self.schema = schema
+        self._init_setup()
+
+    def _send_mattermost_request(self, url: str, method: HttpMethod, **kwargs):
+        headers = kwargs.pop("headers", {})
```

Comment:
Sometime we call it without kwargs
Does that work ok? What will be in kwargs when we call only with url & method?

Reply:
kwargs are used to pass extra arguments to the request (form data, params, body request, files, headers etc)
If we want a plain request (e.g. GET) we do not need to pass any extra arguments